### ISSUE TITLE
ANY alternative?

### ISSUE THREAD

**Issue by itwars** [chzyer/readline#178]
As this project seems to be not more maintained, is there any alternative?

Many thanks

**Comment by quite** [comment]
Mmm, I have some trouble with this readline pkg. Typing beyond end of line, ending up on a new line and then moving back with C-b or M-b screws up the redrawing. The input line keeps moving up the screen (!). Should probably report this, but I'm gonna try https://github.com/peterh/liner.

**Comment by jcburley** [comment]
> I'm gonna try https://github.com/peterh/liner.

[Joker](https://github.com/candid82/joker) has since migrated to its author's own [fork](https://github.com/candid82/liner) of `liner`.

Seems to be working quite well so far!

**Comment by quite** [comment]
I switched to peterh/liner myself. Why the fork of it?

**Comment by jcburley** [comment]
> I switched to peterh/liner myself. Why the fork of it?

Not sure exactly -- I think @candid82 explained it somewhere (maybe Slack), something to do with adding a feature, possibly Bash-style completion (though maybe not, as I think that was just a configurable change)?

**Comment by itwars** [comment]
So it's a fork before a future pull request 👍

**Comment by jcburley** [comment]
> So it's a fork before a future pull request

Unsure, though I don't see any pertinent open PRs. He might have found a pertinent closed PR that was deemed "Won't Fix" and decided to not bother.

Another possibility is that it was to support automatic paren matching, which might be rather peculiar to Joker with its Clojure/Lisp REPL.

At some point I'll probably compare the two myself, but have other deliverables to focus on right now.

**Comment by candid82** [comment]
> Another possibility is that it was to support automatic paren matching, which might be rather peculiar to Joker with its Clojure/Lisp REPL.

Yes, that and also slightly different behavior for alt+backspace (treats `/` as word separator). I might tweak it further to better suit Lisp editing. At some point I might PR my changes but they would have to be configurable to be accepted I think.
